src/ts/content/extensions/enforceTagIndexAccess.ts

```diff
@@ -1,45 +1,55 @@
-import {ConfigKey} from "../../common/entities/config";
-import config from "../../common/entities/config";
+import config, {ConfigKey} from "../../common/entities/config";
 import {createModal} from "../../common/ui/modal";
 import {UIColour} from "../../common/ui/colour";
 import {onLogged} from "./util/onLogged";
-import {isOverlayRendered, removeOverlay} from "../../common/ui/overlay";
+import {clearOverlays} from "../../common/ui/overlay";
 import {invokeWorker} from "../../common/workers/invoker";
 import {WorkerKind} from "../../common/workers/types";
+import {onBackgroundEvent} from "../util/onBackgroundEvent";
+import {BackgroundEvent} from "../../common/backgroundEvent";
 
-window.addEventListener("pageshow", async () => {
+let ready = false;
+
+const enforceTagIndexAccess = async () => {
+	clearOverlays();
 	if (await config.get(ConfigKey.EnforceTagIndexAccess)) {
-		const createWarningModal = () =>
-			createModal({
-				text: "Hey there!",
-				subText: [
-					"It looks like you might not be logged in or do not have a tag index set",
-					"Please login and/or set a Tag Index link under the extension options",
-				],
-				elements: [
-					{
-						kind: "button",
-						text: "Open Options",
-						colour: UIColour.PURPLE,
-						onClick: async () => {
-							invokeWorker(WorkerKind.OpenOptions, null);
-							createWarningModal();
+		const createWarningModal = async () => {
+			ready === false &&
+				createModal({
+					text: "Hey there!",
+					subText: [
+						"It looks like you might not be logged in or do not have a tag index set",
+						"Please login and/or set a Tag Index link under the extension options",
+					],
+					elements: [
+						{
+							kind: "button",
+							text: "Open Options",
+							colour: UIColour.PURPLE,
+							onClick: async () => {
+								void invokeWorker(WorkerKind.OpenOptions, null);
+								return createWarningModal();
+							},
 						},
-					},
-				],
-				colour: UIColour.PURPLE,
-				exitable: false,
-			});
-
-		onLogged({
-			onLogIn: () => {
-				removeOverlay();
-			},
-			onLogOut: async () => {
-				if (!isOverlayRendered()) {
-					createWarningModal();
-				}
-			},
-		});
+					],
+					colour: UIColour.PURPLE,
+					onCancel: createWarningModal,
+				});
+		};
+		return createWarningModal();
 	}
+};
+
+onBackgroundEvent(BackgroundEvent.EditEnforcement, enforceTagIndexAccess);
+window.addEventListener("pageshow", async () => {
+	await onLogged({
+		onLogIn: () => {
+			ready = true;
+			return enforceTagIndexAccess();
```

Comment:
just to check, we know this might be called/put on the dom multiple times, but we're okay with that because we clear all of them before putting a new one?

Reply:
Yeah it could possibly happen 2/3 times but i think it's ok